File: module4/case_study/src/main/java/com/phuong/model/customer/Customer.java
package com.phuong.model.customer;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.phuong.model.contract.Contract;

import javax.persistence.*;
import java.sql.Date;
import java.util.List;

@Entity
public class Customer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String address;

    private Date birthday;

    private String email;

    private Integer gender;

    private String idCard;

    private String name;

    private String phoneNumber;
    
    @ManyToOne
    @JoinColumn(name = "customer_type_id", referencedColumnName = "id")
    private CustomerType customerType;

    @OneToMany(mappedBy = "customer")
    @JsonBackReference
    private List<Contract> contractList;

    public Customer() {
    }

    public Customer(Integer id, String address, Date birthday, String email, Integer gender, String idCard,
                    String name, String phoneNumber, CustomerType customerType, List<Contract> contractList) {
        this.id = id;
        this.address = address;
        this.birthday = birthday;
        this.email = email;
        this.gender = gender;
        this.idCard = idCard;
        this.name = name;
        this.phoneNumber = phoneNumber;
        this.customerType = customerType;
        this.contractList = contractList;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public Date getBirthday() {
        return birthday;
    }

    public void setBirthday(Date birthday) {
        this.birthday = birthday;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public Integer getGender() {
        return gender;
    }

    public void setGender(Integer gender) {
        this.gender = gender;
    }

    public String getIdCard() {
        return idCard;
    }

    public void setIdCard(String idCard) {
        this.idCard = idCard;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPhoneNumber() {
        return phoneNumber;
    }

    public void setPhoneNumber(String phoneNumber) {
        this.phoneNumber = phoneNumber;
    }

    public CustomerType getCustomerType() {
        return customerType;
    }

    public void setCustomerType(CustomerType customerType) {
        this.customerType = customerType;
    }

    public List<Contract> getContractList() {
        return contractList;
    }

    public void setContractList(List<Contract> contractList) {
        this.contractList = contractList;
    }
}

File: module5/angular-test/src/app/model/customer/customer.ts
import {CustomerType} from './customerType';

export class Customer {
  private _id?: number;
  private _address?: string;
  private _birthday?: string;
  private _email?: string;
  private _gender?: number;
  private _idCard?: string;
  private _name?: string;
  private _phoneNumber?: string;
  private _customerType?: CustomerType;

  constructor(id: number, address: string, birthday: string, email: string,
              gender: number, idCard: string, name: string, phoneNumber: string, customerType: CustomerType) {
    this._id = id;
    this._address = address;
    this._birthday = birthday;
    this._email = email;
    this._gender = gender;
    this._idCard = idCard;
    this._name = name;
    this._phoneNumber = phoneNumber;
    this._customerType = customerType;
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get address(): string {
    return this._address;
  }

  set address(value: string) {
    this._address = value;
  }

  get birthday(): string {
    return this._birthday;
  }

  set birthday(value: string) {
    this._birthday = value;
  }

  get email(): string {
    return this._email;
  }

  set email(value: string) {
    this._email = value;
  }

  get gender(): number {
    return this._gender;
  }

  set gender(value: number) {
    this._gender = value;
  }

  get idCard(): string {
    return this._idCard;
  }

  set idCard(value: string) {
    this._idCard = value;
  }

  get name(): string {
    return this._name;
  }

  set name(value: string) {
    this._name = value;
  }

  get phoneNumber(): string {
    return this._phoneNumber;
  }

  set phoneNumber(value: string) {
    this._phoneNumber = value;
  }

  get customerType(): CustomerType {
    return this._customerType;
  }

  set customerType(value: CustomerType) {
    this._customerType = value;
  }
}
